src/Sylius/Bundle/ResourceBundle/Storage/CookieStorage.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Storage;
+
+use Sylius\Component\Resource\Storage\StorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class CookieStorage implements StorageInterface, EventSubscriberInterface
```

Comment:
> It is not my duty to write it for you.

Yeah, but I really would prefer to see a real code instead of some generic concepts being thrown. Talking about code quality and proving that some approach is cleaner than the another one without hard evidence is not enough.

I'd like to see the code for the reasons mentioned before, there is a lot of glue needed between these two parts of cookie storage so I'm interested how would you deal with it.

> Saying, that this class has one responsibility is like putting all the code into the controller and saying, that this class has also only one responsibility: generate the response.

That's true at a given level of abstraction and we all rely on that abstraction successfully since the first days of Symfony or other request-response frameworks.

> This class updates a storage on each request as well as providing a cookie storage behaviour (I agree, that it is not a typical one).

Updating storage on master request and updating the response with new values are one of the parts of providing cookie storage behaviour, it would not be fulfilled without them.